woodwork/type_sys/type_system.py

```diff
@@ -154,41 +166,57 @@ def add_type(self, logical_type, inference_function=None, parent=None):
-        if parent:
-            self.update_relationship(logical_type, parent)
-
-    def remove_type(self, logical_type):
+        elif ltype_is_registered and treatment.lower() == "replace":
+            self.remove_type(logical_type=logical_type)
+        if not ltype_is_registered or treatment.lower() == "replace":
+            self.update_inference_function(logical_type, inference_function)
+            if parent:
+                self.update_relationship(logical_type, parent)
+
+    def remove_type(self, logical_type: LogicalType, treatment: str = None):
         """Remove a logical type from the TypeSystem. Any children of the remove type will have their parent
         set to the parent of the removed type.
 
         Args:
             logical_type (LogicalType): The LogicalType to remove.
+            treatment (str, optional): If the specified logical type doesn't exist, determine the logic that should
+                be applied. Options are "ignore" and None (default). "Ignore" will not raise an error if the passed
+                LogicalType doesn't exists. The default value of None will raise an error.
         """
         if isinstance(logical_type, str):
             logical_type = self.str_to_logical_type(logical_type)
         self._validate_type_input(logical_type=logical_type)
         # Remove the inference function
         if logical_type == self.default_type:
             raise ValueError("Default LogicalType cannot be removed")
-        self.inference_functions.pop(logical_type)
-
-        # If the removed type had children we need to update them
-        children = self._get_children(logical_type)
-        if children:
-            parent = self._get_parent(logical_type)
-            for child in children:
-                self.update_relationship(child, parent)
 
-        # Rebuild the relationships list to remove any reference to the removed type
-        self.relationships = [
-            rel for rel in self.relationships if logical_type not in rel
-        ]
+        registered_ltype_names = {ltype.__name__ for ltype in self.registered_types}
+        ltype_is_registered = logical_type.__name__ in registered_ltype_names
+        if ltype_is_registered:
+            self.inference_functions.pop(logical_type)
+
+            # If the removed type had children we need to update them
+            children = self._get_children(logical_type)
+            if children:
+                parent = self._get_parent(logical_type)
+                for child in children:
+                    self.update_relationship(child, parent)
+
+            # Rebuild the relationships list to remove any reference to the removed type
+            self.relationships = [
+                rel for rel in self.relationships if logical_type not in rel
+            ]
+        else:
+            if treatment is None or treatment.lower() != "ignore":
+                raise ValueError(
```

Comment:
Everything looks good to me! but should definitely have more a few more eyes look at this before merging.